### PR TITLE
Variable View Editor Button

### PR DESCRIPTION
**Pull request by IanMatthewHuff**
For #4486 

-   [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [x] Title summarizes what is changing.
-   [x] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

**Comment by IanMatthewHuff** [comment]
Per Claudia convo button still there, but disabled when view is currently visible.

File: src/client/common/application/types.ts [hunk 1]
```diff
@@ -1093,7 +1093,10 @@ export interface IWebview {
 
 // Wraps the VS Code webview view
 export const IWebviewView = Symbol('IWebviewView');
-export interface IWebviewView extends IWebview {}
+export interface IWebviewView extends IWebview {
+    readonly onDidChangeVisiblity: Event<void>;
```
Comment (IanMatthewHuff) [inline src/client/common/application/types.ts:1097]: vscode webviewview doesn't handle visibility the same as vscode webviewpanel, so this is handled specific to the IWebviewView interface.

File: src/client/common/application/webviewViews/webviewView.ts [hunk 2]
```diff
@@ -38,5 +49,11 @@ export class WebviewView extends Webview implements IWebviewView {
                 this.panelOptions.listener.onMessage(message.type, message.payload);
             })
         );
+
+        this.disposableRegistry.push(
+            webviewHost.onDidChangeVisibility(() => {
```
Comment (rchiodo) [inline src/client/common/application/webviewViews/webviewView.ts:54]: Does this fire when the host is launched? Seems like it should if it doesn't.

**Review by rchiodo — APPROVED** [review]

File: src/client/common/application/webviewViews/webviewView.ts (hunk 2 above)
Comment (IanMatthewHuff) [reply src/client/common/application/webviewViews/webviewView.ts:54]: @rchiodo Yeah, I think you are right. I don't believe that I noticed as my variable view does an initial direct check for visible state before it hooks up and watches for the event. But we should also fire an initial fire of this.

**Pull request merged**